### PR TITLE
[HDRP] APV Baking and Settings window

### PR DESCRIPTION
**Pull request by alelievr**
### Purpose of this PR
Add a new window to manage settings for APV. The new window is located in **Window > Rendering > Probe Volume Settings (Experimental)**

image

---
### Testing status
- Tested undo/redo on all the UI
- Tested the bake / load scene / clear buttons
- Multi-scene baking
- Tested that parameters in the UI actually works

### CODE REVIEW

File: com.unity.render-pipelines.core/Editor/Lighting/ProbeVolume/ProbeGIBaking.cs [hunk 1]
```diff
@@ -320,6 +320,9 @@ internal static void PerformDilation()
 
             if (dilationSettings.dilationDistance > 0.0f)
             {
+                // Make sure all pending unloads are done before we reset everything through ForceSHBands.
```
Comment (JulienIgnace-Unity) [inline com.unity.render-pipelines.core/Editor/Lighting/ProbeVolume/ProbeGIBaking.cs:323]: @FrancescoC-unity can you validate this fix?
Basically some scenes are still in the pending unload list, we then reset the ref volume to a different size which makes subsequent unloading fail (out of bounds for flat array access).

Comment (FrancescoC-unity) [reply com.unity.render-pipelines.core/Editor/Lighting/ProbeVolume/ProbeGIBaking.cs:323]: Uhm this is potentially dangerous I think, as we are flushing also the loading that are pending while we want to flush them only after we force the sh bands. 

We might need something that only flushes the unloading.

Comment (JulienIgnace-Unity) [reply com.unity.render-pipelines.core/Editor/Lighting/ProbeVolume/ProbeGIBaking.cs:323]: Right, that makes sense. Shouldn't we be flushing everything at the start of baking anyway?
There's probably a better place for this I guess.

Comment (FrancescoC-unity) [reply com.unity.render-pipelines.core/Editor/Lighting/ProbeVolume/ProbeGIBaking.cs:323]: Yes we can try to do that, we can try at the start of OnAdditionalProbesBakeCompleted , worth testing this a bit as I wonder if there are unwanted side effects we are not thinking about.

File: com.unity.render-pipelines.core/Editor/Lighting/ProbeVolume/ProbeVolumeBakingWindow.cs [hunk 2]
```diff
@@ -0,0 +1,525 @@
...
+            m_ScenesInSet = new ReorderableList(set.sceneGUIDs, typeof(string), true, true, true, true);
+            m_ScenesInSet.drawHeaderCallback = (rect) => EditorGUI.LabelField(rect, "Scenes", EditorStyles.largeLabel);
+            m_ScenesInSet.multiSelect = true;
+            m_ScenesInSet.drawElementCallback = (rect, index, active, focused) =>
+            {
+                var guid = set.sceneGUIDs[index];
+                // Find scene name from GUID:
+                var scene = FindSceneData(guid);
+
+                // TODO: Add a label on the first scene to say the the lighting settings of this scene will be used for baking
```
Comment (FrancescoC-unity) [inline com.unity.render-pipelines.core/Editor/Lighting/ProbeVolume/ProbeVolumeBakingWindow.cs:225]: We do have it now, so can get rid of this TODO.

File: com.unity.render-pipelines.core/Editor/Lighting/ProbeVolume/ProbeVolumeBakingWindow.cs [hunk 3]
```diff
@@ -0,0 +1,525 @@
...
+                    path = path,
+                    guid = s
+                };
+            }).ToList();
+        }
+
+        SceneData FindSceneData(string guid)
+        {
+            // TODO: replace m_ScenesInProject list by a dictionary
+            var data = m_ScenesInProject.FirstOrDefault(s => s.guid == guid);
```
Comment (FrancescoC-unity) [inline com.unity.render-pipelines.core/Editor/Lighting/ProbeVolume/ProbeVolumeBakingWindow.cs:199]: Do we want to this TODO before getting this through?

File: com.unity.render-pipelines.core/Editor/Lighting/ProbeVolume/ProbeVolumeBakingWindow.cs [hunk 4]
```diff
@@ -0,0 +1,525 @@
...
+            };
+
+            m_BakingSets.onRemoveCallback = (list) =>
+            {
+                if (m_BakingSets.count == 1)
+                {
+                    EditorUtility.DisplayDialog("Can't delete baking set", "You can't delete the last Baking set. You need to have at least one.", "Ok");
+                    return;
+                }
+                if (EditorUtility.DisplayDialog("Delete selected baking set?", $"Do you really want to delete the baking set '{sceneData.bakingSets[list.index].name}'?", "Yes", "Cancel"))
```
Comment (FrancescoC-unity) [inline com.unity.render-pipelines.core/Editor/Lighting/ProbeVolume/ProbeVolumeBakingWindow.cs:158]: Delete _the_ selected baking set

File: com.unity.render-pipelines.core/Editor/Lighting/ProbeVolume/ProbeVolumeBakingWindow.cs (hunk 3 above)
Comment (alelievr) [reply com.unity.render-pipelines.core/Editor/Lighting/ProbeVolume/ProbeVolumeBakingWindow.cs:199]: I guess it's fine

**Pull request merged**